egg/core/callbacks.py

```diff
@@ -207,3 +217,115 @@ def on_test_end(self, loss: float, logs: Interaction, epoch: int):
     def on_epoch_end(self, loss: float, logs: Interaction, epoch: int):
         if epoch in self.train_epochs:
             self.dump_interactions(logs, "train", epoch, self.folder_path)
+
+
+class ProgressBarLogger(Callback):
+    class CompletedColumn(ProgressColumn):
+        def render(self, task):
+            """Calculate common unit for completed and total."""
+            download_status = f"{int(task.completed)}/{int(task.total)} btc"
+            return Text(download_status, style="progress.download")
+
+    class TransferSpeedColumn(ProgressColumn):
+        """Renders human readable transfer speed."""
+
+        def render(self, task):
+            """Show data transfer speed."""
+            speed = task.speed
+            if speed is None:
+                return Text("?", style="progress.data.speed")
+            speed = f"{speed:,.{2}f}"
+            return Text(f"{speed} btc/s", style="progress.data.speed")
+
+    def __init__(
+        self,
+        n_epochs,
+        train_data_len=None,
+        eval_data_len=None,
+    ):
+        self.n_epochs = n_epochs
+        self.cur_epoch = 1
```

Comment:
Wouldn't this cause an odd behaviour if have have a progress bar when resuming a training run?
I would set one as the default parameter but add the option to pass a custom cur_epoch

Reply:
fixed in 9999cee